pkg/api/validation/dynakube/oneagent.go

```diff
@@ -23,6 +23,8 @@ Use a nodeSelector to avoid this conflict. Conflicting DynaKubes: %s`
 
 	errorVolumeStorageReadOnlyModeConflict = `The DynaKube specification specifies a read-only host file system while OneAgent has volume storage enabled.`
 
+	errorPublicImageWithWrongConfig = `The DynaKube specification specifies a custom (and therefor probably a public) image in combination with a mode that needs write permissions for volume mounts.`
```

Comment:
```suggestion
	errorPublicImageWithWrongConfig = `Custom OneAgent image is only supported when CSI Driver is used.`
```
It would be much nicer to have a guide in the official documentation to link to explaining the feature flag, but for now I think we can boil it down to this.